WordPress/Classes/ViewRelated/Blog/Blog + Me/UIBarButtonItem+MeBarButton.swift

```diff
@@ -3,47 +3,52 @@ import UIKit
 
 /// Add a UIBarButtonItem to the navigation bar that  presents the Me scene.
 extension UIViewController {
-    @objc
-    func addMeButtonToNavigationBar(email: String?, meScenePresenter: ScenePresenter) {
-        navigationItem.rightBarButtonItem = UIBarButtonItem(
+
+    @objc func addMeButtonToNavigationBar(email: String?, meScenePresenter: ScenePresenter? = nil) {
```

Comment:
This is an existing method that adds the "Me" bar button item to a navigation bar. We need this component for the Migration Screens as well.

The problem was that this button was always displaying the "Me" scene when tapped and there wasn't a way to disable this behavior.

The changes I added are minimal to avoid regression bugs, and they are:
- Disable the tapping behavior by making some params optionals
- Expose this `UIBarButtonItem` convenience initializer so that we can mutates its [default accessibility values](https://github.com/wordpress-mobile/WordPress-iOS/pull/19533/files#diff-0adecdb95492cde4e691510d54b24aa1df7deca52b2193f84496be9b2929fd59R26).